### PR TITLE
OAuth Authenticator updates

### PR DESCRIPTION
**Pull request by MrSmoke**
- Rename `Helpers` to `TokenHelpers` and change from `internal` to `public`
- Add `GetResourceOwnerPasswordTokenAsync` to `TokenClient`
- Add `HasExpired` extension method to `Token` to check for expired tokens
  - Move the clock drift logic to when checking if expired instead of removing from the actual expire time stored
- Add missing configure awaits for old .net
- Use `Array.Empty` over `new List<>`

### CODE REVIEW

**Comment by MrSmoke** [comment]
@happymancoder 
> [17:32] looks to good to me

File: src/RestClient/Authenticators/OAuth/src/Tokens/TokenExtensions.cs [hunk 1]
```diff
@@ -4,15 +4,17 @@
 
     public static class TokenExtensions
     {
-        public static bool IsValid(this Token token)
+        public static bool HasExpired(this Token token)
         {
-            if (token == null)
-                return false;
+            if (token is null)
```
Comment (eyamenko) [inline src/RestClient/Authenticators/OAuth/src/Tokens/TokenExtensions.cs:9]: nit: https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception.throwifnull?view=net-8.0#system-argumentnullexception-throwifnull(system-object-system-string)

File: src/RestClient/Authenticators/OAuth/src/Tokens/TokenExtensions.cs [hunk 2]
```diff
@@ -4,15 +4,17 @@
 
     public static class TokenExtensions
     {
-        public static bool IsValid(this Token token)
+        public static bool HasExpired(this Token token)
```
Comment (eyamenko) [inline src/RestClient/Authenticators/OAuth/src/Tokens/TokenExtensions.cs:7]: Unit tests? Docs?

File: src/RestClient/Authenticators/OAuth/src/TokenHelpers.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿namespace ClickView.Extensions.RestClient.Authenticators.OAuth;
+
+using System;
+using IdentityModel.Client;
+using Tokens;
+
+public static class TokenHelpers
```
Comment (eyamenko) [inline src/RestClient/Authenticators/OAuth/src/TokenHelpers.cs:7]: Unit tests? Docs?

**Pull request merged**